scipy/stats/tests/test_multivariate.py

```diff
@@ -1635,6 +1643,248 @@ def test_haar(self):
+        samples = dist.rvs(size=100000)
+        _, p = normaltest(samples)
+        assert (p > P_VAL_MIN)
+
+        dist = multivariate_t([-2, 3], [[10, -1], [-1, 10]], df=100000,
+                              seed=42)
+        samples = dist.rvs(size=100000)
+        _, p = normaltest(samples)
+        assert ((p > P_VAL_MIN).all())
+
+    @patch('scipy.stats.multivariate_normal._logpdf')
+    def test_mvt_with_high_df_is_normal(self, mock):
+        dist = multivariate_t(0, 1, df=np.inf, seed=7)
+        assert isinstance(dist, multivariate_normal_frozen)
+        multivariate_t.pdf(0, df=np.inf)
+        assert mock.call_count == 1
+        multivariate_t.logpdf(0, df=np.inf)
+        assert mock.call_count == 2
+
+    def test_default_arguments(self):
+        dist = multivariate_t()
+        assert_equal(dist.loc, [0])
+        assert_equal(dist.shape, [[1]])
+        assert (dist.df == 1)
+
+    def test_argument_handling(self):
+        dist = multivariate_t(loc=None, shape=None, df=None)
+        assert_equal(dist.loc, 0)
+        assert_equal(dist.shape, 1)
+        assert (dist.df == 1)
+
+        dist = multivariate_t(loc=None, shape=None, df=7)
+        assert_equal(dist.loc, 0)
+        assert_equal(dist.shape, 1)
+        assert (dist.df == 7)
+
+        dist = multivariate_t(loc=None, shape=[[7, 0], [0, 7]], df=None)
+        assert_equal(dist.loc, [0, 0])
+        assert_equal(dist.shape, [[7, 0], [0, 7]])
+        assert (dist.df == 1)
+
+        dist = multivariate_t(loc=None, shape=[[7, 0], [0, 7]], df=7)
+        assert_equal(dist.loc, [0, 0])
+        assert_equal(dist.shape, [[7, 0], [0, 7]])
+        assert (dist.df == 7)
+
+        dist = multivariate_t(loc=[7, 7], shape=None, df=None)
+        assert_equal(dist.loc, [7, 7])
+        assert_equal(dist.shape, [[1, 0], [0, 1]])
+        assert (dist.df == 1)
+
+        dist = multivariate_t(loc=[7, 7], shape=None, df=7)
+        assert_equal(dist.loc, [7, 7])
+        assert_equal(dist.shape, [[1, 0], [0, 1]])
+        assert (dist.df == 7)
+
+        dist = multivariate_t(loc=[7, 7], shape=[[7, 0], [0, 7]], df=None)
+        assert_equal(dist.loc, [7, 7])
+        assert_equal(dist.shape, [[7, 0], [0, 7]])
+        assert (dist.df == 1)
+
+        dist = multivariate_t(loc=[7, 7], shape=[[7, 0], [0, 7]], df=7)
+        assert_equal(dist.loc, [7, 7])
+        assert_equal(dist.shape, [[7, 0], [0, 7]])
+        assert (dist.df == 7)
+
+    def test_argument_error_handling(self):
+        # `loc` should be a one-dimensional vector.
+        with pytest.raises(ValueError):
+            multivariate_t(loc=[[1, 1]])
+
+        # `shape` should be scalar or square matrix.
+        with pytest.raises(ValueError):
+            multivariate_t(shape=[[1, 1], [2, 2], [3, 3]])
+
+        # Confirm df > 0, otherwise `ValueError`:
+        with pytest.raises(ValueError):
+            multivariate_t(df=-1)
+        with pytest.raises(ValueError):
+            multivariate_t(df=0)
```

Comment:
these could be `assert_raises`, would be less code and context managers do not add much here.

Reply:
Sure, I've switched to `assert_raises`. I suppose this fits the paradigm in this file, but FWIW both [this StackOverflow](https://stackoverflow.com/questions/23337471/) and the [Pytest docs](https://docs.pytest.org/en/stable/assert.html#assertraises) suggest the context manager approach. I prefer it because you see the function call directly, rather than implicitly, e.g.:

```python
assert_raises(ValueError,
              multivariate_t,
              **dict(loc=loc, shape=shape, df=df))
```